ethcore/src/machine.rs

```diff
@@ -193,6 +193,30 @@ impl EthereumMachine {
 						.and_then(|b| state.transfer_balance(child, beneficiary, &b, CleanupMode::NoEmpty))?;
 				}
 			}
+
+			if let Some(irregular_changes) = self.params.irregular_state_changes.get(&block.header().number()) {
+				let state = block.state_mut();
+				for &(address, ref irregular_account) in irregular_changes {
+					match irregular_account {
+						&IrregularStateChangeAccount::Set {
```

Comment:
single arm match can be replaced by an irrefutable destructuring.
`let &IrregularStateChangAccount { none, balance, ref code, ref storage } = irregular_account;`

Reply:
> if the IrregularStateChangeAccount enum changed to have more variants, it cannot detect it and throw type errors.

that's not true. only irrefutable pattern matches can be used in a let-binding. adding another variant would make that pattern match no longer irrefutable.